ambry-network/src/main/java/com.github.ambry.network/NetworkClient.java

```diff
@@ -49,8 +49,7 @@
   private final HashMap<String, RequestMetadata> connectionIdToRequestInFlight;
   private final AtomicLong numPendingRequests;
   private final int checkoutTimeoutMs;
-  // @todo: this needs to be empirically determined.
-  private final int POLL_TIMEOUT_MS = 1;
+  private final int POLL_TIMEOUT_MS = 300;
```

Comment:
i don't think this should be 300. I think it should be set to the value of request timeouts and therefore should be configurable when starting up the network client.